From fb2fa8348d87024d94905ed94f9f78149726df88 Mon Sep 17 00:00:00 2001
From: Marek Kasik <mkasik@redhat.com>
Date: Fri, 25 Jan 2019 16:41:24 +0100
Subject: [PATCH 13/25] printing: Don't show Rejecting Jobs when we don't know

Set reasonable default values for printers discovered
by Avahi which do not have 'printer-type' attribute.
This is the case for network printers which were not
published by CUPS.

Related to the issue #1509.
---
 modules/printbackends/cups/gtkprintbackendcups.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/modules/printbackends/cups/gtkprintbackendcups.c b/modules/printbackends/cups/gtkprintbackendcups.c
index d81cde8355..800959a1e5 100644
--- a/modules/printbackends/cups/gtkprintbackendcups.c
+++ b/modules/printbackends/cups/gtkprintbackendcups.c
@@ -2913,6 +2913,9 @@ create_cups_printer_from_avahi_data (AvahiConnectionTestData *data)
   info->avahi_printer = TRUE;
   info->printer_name = data->printer_name;
   info->printer_uri = data->printer_uri;
+  info->default_printer = FALSE;
+  info->remote_printer = TRUE;
+  info->is_accepting_jobs = TRUE;
 
   if (data->got_printer_state)
     {
-- 
2.20.1

